addons/sale_pdf_quote_builder/wizards/sale_pdf_quote_builder_dynamic_fields_wizard.py

```diff
@@ -0,0 +1,134 @@
+        for wizard in self:
+            wizard.dynamic_fields_wizard_line_ids = [Command.clear()]
+            form_field_path_map = json.loads(self.env['ir.config_parameter'].sudo().get_param(
+                'sale_pdf_quote_builder.form_field_path_mapping', DEFAULT_FORM_FIELD_PATH_MAPPING
+            ))
+            current_form_fields = json.loads(
+                wizard.current_form_fields
+            ) if wizard.current_form_fields else {}
+
+            # Get the form fields not yet in the config parameters
+            lines_vals = []
+            for doc_type, form_fields in current_form_fields.items():
+                for form_field in form_fields:
+                    if form_field not in form_field_path_map[doc_type]:
+                        lines_vals.extend([
+                            {'document_type': doc_type, 'form_field': form_field, 'path': ''}
+                        ])
+            # Add existing values from the config parameter
+            lines_vals.extend([
+                {'document_type': doc_type, 'form_field': form_field, 'path': path}
+                for doc_type, mapping in form_field_path_map.items()
+                for form_field, path in mapping.items()
+            ])
+            # Create the wizard lines
+            wizard.dynamic_fields_wizard_line_ids = [Command.create(vals) for vals in lines_vals]
+
+    def validate_dynamic_fields_configuration(self):
+        """ Save this mapping configuration in the config parameters. """
+        for wizard in self:
+            wizard_lines = wizard.dynamic_fields_wizard_line_ids
+            so_wiz_line = wizard_lines.filtered(lambda line: line.document_type == 'header_footer')
+            sol_wiz_line = wizard_lines - so_wiz_line
+            new_mapping = {
+                'header_footer': {line.form_field: line.path or '' for line in so_wiz_line},
+                'product_document': {line.form_field: line.path or '' for line in sol_wiz_line},
+            }
+
+            self.env['ir.config_parameter'].set_param(
+                'sale_pdf_quote_builder.form_field_path_mapping', json.dumps(new_mapping)
+            )
+
+
+class SalePDFQuoteBuilderDynamicFieldsWizardLine(models.TransientModel):
+    _name = 'sale.pdf.quote.builder.dynamic.fields.wizard.line'
+    _description = "SalePdfQuoteBuilderDynamicFieldsWizardLine transient representation"
+
+    dynamic_fields_wizard_id = fields.Many2one(
+        'sale.pdf.quote.builder.dynamic.fields.wizard', required=True, ondelete='cascade'
+    )
+    document_type = fields.Selection(
+        string="Document Type",
+        selection=[('header_footer', "Header/Footer"), ('product_document', "Product Document")],
+        default='header_footer',
+        required=True,
+    )
+    form_field = fields.Char(
+        string="Form Field", help="Form Field Name as seen in the PF form", required=True,
+    )
+    path = fields.Char(default='', help="Odoo path to get the expected value to fill the PDF.")
```

Comment:
is the default really necessary ? :thinking:

Reply:
Yes, for the next task, where the difference between config.get(field) == "" would mean that the form field is accepted and needs to be manually field whereas config.get(field) == None would mean it's not authorized.
This default is only used when adding new lines, not there in the doc, as those are directly created using an empty string.